pin.go

```diff
@@ -79,153 +76,13 @@ func computePinNVIndexPostInitAuthPolicies(alg tpm2.HashAlgorithmId, updateKeyNa
 	return out, nil
 }
 
-// createPinNVIndex creates a NV index that is associated with a sealed key object and is used for implementing PIN support. It is
-// also used as a counter to support revoking of dynamic authorization policies.
+// performPinChangeV0 changes the authorization value of the dynamic authorization policy counter associated with the public
+// argument, for PIN integration in version 0 key files. This requires the authorization policy digests initially returned from
+// (the now removed) createPinNVIndex function in order to execute the policy session required to change the authorization value.
+// The current authorization value must be provided via the oldAuth argument.
 //
-// To prevent someone with knowledge of the owner authorization (which is empty unless someone has taken ownership of the TPM) from
-// resetting the PIN by just undefining and redifining a new NV index with the same properties, we need a way to prevent someone
-// from being able to create an index with the same name. To do this, we require the NV index to have been written to and only allow
-// the initial write with a signed authorization policy. Once initialized, the signing key that authorized the initial write is
-// discarded. This works because the name of the signing key is included in the authorization policy digest for the NV index, and the
-// authorization policy digest and attributes are included in the name of the NV index. Without the private part of the signing key,
-// it is impossible to create a new NV index with the same name, and so, if this NV index is undefined then it becomes impossible to
-// satisfy the authorization policy for the sealed key object to which it is associated.
-//
-// The NV index will be created with an authorization policy that permits TPM2_NV_Read and TPM2_PolicyNV without knowing the PIN,
-// and an authorization policy that permits TPM2_NV_Increment with a signed authorization policy, signed by the key associated with
-// updateKeyName.
-func createPinNVIndex(tpm *tpm2.TPMContext, handle tpm2.Handle, updateKeyName tpm2.Name, hmacSession tpm2.SessionContext) (*tpm2.NVPublic, tpm2.DigestList, error) {
-	initKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	if err != nil {
-		return nil, nil, xerrors.Errorf("cannot create signing key for initializing NV index: %w", err)
-	}
-
-	initKeyPublic := createPublicAreaForECDSAKey(&initKey.PublicKey)
-	initKeyName, err := initKeyPublic.Name()
-	if err != nil {
-		return nil, nil, xerrors.Errorf("cannot compute name of signing key for initializing NV index: %w", err)
-	}
-
-	nameAlg := tpm2.HashAlgorithmSHA256
-
-	// The NV index requires 5 policies:
```

Comment:
we should keep this comment somewhere close to v0 metadata otherwise it's hard to follow version == 0 auth code, part of it is kept implemented by computeV0PinNVIndexPostInitAuthPolicies? but that is used only validate so it's a bit too indirect

Reply:
Would it be ok to just reference the comment for computeV0PinNVIndexPostInitAuthPolicies from other places?

Reply:
given you have been very explicit about that now, seems ok